src/ethereum/cancun/fork.py

```diff
@@ -749,9 +759,7 @@ def process_transaction(
     for address in output.accounts_to_delete:
         destroy_account(env.state, address)
 
-    for address in output.touched_accounts:
-        if account_exists_and_is_empty(env.state, address):
-            destroy_account(env.state, address)
+    destroy_touched_empty_accounts(env.state, output.touched_accounts)
```

Comment:
@SamWilsn Had missed using the re-factored function to destroy the empty accounts. This is the only update I've just made.